go/src/vendors/sonic/client/dbclient/redis_provider.go

```diff
@@ -0,0 +1,87 @@
+package dbclient
+
+import (
+        . "go/src/lib/lomcommon"
+	"github.com/go-redis/redis"
+)
+
+const (
+	redisServerAddr     = "localhost:6379"
+	redisServerPassword = ""
+	COUNTER_DB_ID       = 2
+)
+
+var dbToRedisClientMapping map[int]*redis.Client
+
+func init() {
+	dbToRedisClientMapping = make(map[int]*redis.Client)
+}
+
+/*
+Gets redis connection for a specific redis database. The connection is re-used.
+If multiple go routines call this at same time, the last client will be stored.
+*/
+func GetRedisConnectionForDatabase(databaseIdentifier int) (*redis.Client, error) {
+	if !validateDbIdentifier(databaseIdentifier) {
+		return nil, LogError("invalid databaseIdentifier (%d)", databaseIdentifier)
```

Comment:
LogError prints message. May be its better to just return this messgae as error object and let caller handle it as its a wrapper code.

Reply:
Done.